common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -2276,22 +2276,25 @@ bool ap_wlan_hal_dwpal::process_dwpal_event(char *buffer, int bufLen, const std:
 
         char MACAddress[MAC_ADDR_SIZE] = {0};
         int status_code                = 0;
+        char vap_name[beerocks::message::IFACE_NAME_LENGTH] = {0};
         size_t numOfValidArgs[4]       = {0};
         FieldsToParse fieldsToParse[]  = {
             {NULL /*opCode*/, &numOfValidArgs[0], DWPAL_STR_PARAM, NULL, 0},
-            {NULL, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL, 0},
+            {(void*) vap_name, &numOfValidArgs[1], DWPAL_STR_PARAM, NULL,
+             beerocks::message::IFACE_NAME_LENGTH},
             {(void *)MACAddress, &numOfValidArgs[2], DWPAL_STR_PARAM, NULL, sizeof(MACAddress)},
             {(void *)&status_code, &numOfValidArgs[3], DWPAL_INT_PARAM, "status_code=", 0},
             /* Must be at the end */
             {NULL, NULL, DWPAL_NUM_OF_PARSING_TYPES, NULL, 0}};
 
-        if (dwpal_string_to_struct_parse(buffer, bufLen, fieldsToParse, sizeof(MACAddress)) ==
-            DWPAL_FAILURE) {
+        if (dwpal_string_to_struct_parse(buffer, bufLen, fieldsToParse,
+                                         sizeof(fieldsToParse)) == DWPAL_FAILURE) {
```

Comment:
If you look to the piece of INTERCAL code that is dwpal, you'll see that the `field` member of the fieldsToParse argument is assumed to point to a struct member of the first element of an array-of-structs, and an intricate way to avoid using the offsetof operator. The userBufLen argument is supposed to be th total size of that array-of-structs. So indeed, if you're not using an array-of-structs, you have to manually add up the size of the different fields.

Reply:
I saw the same today with Denis and we also didn't get why we pass a partial size.